### PR TITLE
checkConditional: number conditional work only for number or strings

### PR DESCRIPTION
**Pull request by tahini**
fixes #1608

The type of the response should be `number` or `string` for correct number comparison. Add a tests with a `null` value and compare with 0, as `Number(null)` converts to 0, but the 0 equality conditional should fail for null values.

## Summary by CodeRabbit

* **Tests**
  * Expanded conditional comparison coverage for numeric-string inputs and numeric coercion edge cases.
  * Added assertions ensuring the null sentinel (`null`) compared to `0` using strict equality evaluates to `false`.

* **Bug Fixes**
  * Improved numeric conditional evaluation to correctly handle numeric responses provided as strings (including trimming and finite-number validation).
  * Refined equality and relational comparisons to more reliably distinguish `null` from `0` and compare numeric values consistently.

### CODE REVIEW

File: packages/evolution-common/src/services/widgets/conditionals/checkConditionals.ts [hunk 1]
```diff
@@ -116,8 +116,8 @@ export const checkConditionals = ({
                 conditionMet = false;
                 break;
             }
-        } else if (typeof value === 'number') {
-            // For Number
+        } else if (typeof value === 'number' && (typeof response === 'number' || typeof response === 'string')) {
```
Comment (greenscientist) [inline packages/evolution-common/src/services/widgets/conditionals/checkConditionals.ts:119]: Should we have some explicit conversion for the case we have a string

File: packages/evolution-common/src/services/widgets/conditionals/checkConditionals.ts [hunk 2]
```diff
@@ -116,8 +116,8 @@ export const checkConditionals = ({
...
+            // For Number values, when response is string or number
             switch (comparisonOperator) {
             case '===':
                 conditionMet = Number(response) === value;
```
[the remark this answers is not in the document]
Comment (kaligrafy) [reply packages/evolution-common/src/services/widgets/conditionals/checkConditionals.ts]: Effectivement

**Review by kaligrafy — CHANGES_REQUESTED** [review]

**Review by kaligrafy — APPROVED** [review]
